### PR TITLE
Add: a plugin results statistic

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

* Default: Summary will be printed after run
  * Will not run if `--no-statistic` argument will used
* Prints the statistic how many errors per plugin have been found   

Summary printing:
```
  Plugin                                   Error Count
  ----------------------------------------------------
× check_updated_date_version                       830
× check_includes                                     8
× check_todo_tbd                                     2
  ----------------------------------------------------
× sum                                              840
```

**Why**:

**How**:

**Checklist**:

- [ ] Tests
- [ ] Conventional commit message
- [ ] Documentation

### CODE REVIEW

File: troubadix/runner.py [hunk 1]
```diff
@@ -99,6 +120,32 @@ def _report_bold_info(self, message: str):
     def _report_ok(self, message: str):
         self._term.ok(message)
 
+    def _process_plugin_results(self, results: OrderedDict):
+
```
Comment (bjoernricks) [inline troubadix/runner.py:124]: IMHO it is important to specify the internal types of the Dict. And also the method doesn't care about the OrderedDict "interface" here and just expects a dict. Btw. dicts are ordered too nowadays.
```suggestion
    def _process_plugin_results(self, results: Dict[str, List[LinterMessage]]) -> None:
```

**Review by bjoernricks — CHANGES_REQUESTED** [review]

File: troubadix/runner.py (hunk 1 above)
Comment (y0urself) [reply troubadix/runner.py:124]: You have introduced the `OrderedDict` in `FileResults` ;)

**Review by Dexus — APPROVED** [review]

**Review by Dexus — APPROVED** [review]

File: troubadix/runner.py [hunk 2]
```diff
@@ -84,21 +105,48 @@ def _report_results(self, results: List[LinterMessage]):
...
+            plugin_results,
+        ) in results.items():
+            if plugin_results[0] or self.debug:
+                self._report_info(f"Running plugin {plugin_name}")
+
+            # add the results to the statistic
+            self.result_counts.add_result_counts(plugin_name, plugin_results[1])
+
+            with self._term.indent():
+                self._report_results(plugin_results[0])
```
Comment (bjoernricks) [inline troubadix/runner.py:138]:
```suggestion
            if plugin_result or self.debug:
                self._report_info(f"Running plugin {plugin_name}")

            # add the results to the statistic
            self.result_counts.add_result_counts(plugin_name, len(plugin_results))

            with self._term.indent():
                self._report_results(plugin_results)
```

**Review by bjoernricks — CHANGES_REQUESTED** [review]
We should only add the result counts where it is actually used.

File: troubadix/runner.py [hunk 3]
```diff
@@ -58,6 +59,19 @@ def add_plugin_results(
...
+    def __init__(self):
+        self.result_counts = {}
+
+    def add_result_counts(self, plugin: str, count: int):
+        """Add the number of results (count) to the dict for the
+        plugin name (plugin)"""
+        if count > 1:
+            if plugin not in self.result_counts:
+                self.result_counts[plugin] = 0
+            self.result_counts[plugin] += count
```
Comment (bjoernricks) [inline troubadix/runner.py:72]:
```suggestion
    def __init__(self):
        self.result_counts = defaultdict(int)

    def add_result_counts(self, plugin: str, count: int):
        """Add the number of results (count) to the dict for the
        plugin name (plugin)"""
        if count > 1:
            self.result_counts[plugin] += count
```

**Review by bjoernricks — CHANGES_REQUESTED** [review]
Simplified ResultCounts class by using [defaultdict](https://docs.python.org/3.9/library/collections.html#collections.defaultdict) from collections module.

File: troubadix/runner.py [hunk 4]
```diff
@@ -84,21 +100,50 @@ def _report_results(self, results: List[LinterMessage]):
...
+    def _process_plugin_results(
+        self, results: Dict[str, List[LinterMessage]]
+    ) -> None:
+        # print the files plugin results
+        for (
+            plugin_name,
+            plugin_results,
+        ) in results.items():
+            if plugin_results or self.debug:
+                self._report_info(f"Running plugin {plugin_name}")
```
Comment (bjoernricks) [inline troubadix/runner.py:127]:
```suggestion
                self._report_info(f"Results for plugin {plugin_name}")
```

**Review by bjoernricks — APPROVED** [review]

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**